site/docs/v3-adoption-progress.md

```diff
@@ -0,0 +1,173 @@
+---
+title: "Apache Iceberg V3 Table Format Adoption Progress"
+---
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Apache Iceberg V3 Table Format Adoption Progress
+
+This document tracks the implementation status of Apache Iceberg V3 Table Format features across different engines and components.
+
+## Overview
+
+Apache Iceberg V3 Table Format introduces several new features including new data types, row lineage, deletion vectors, default values, and table encryption. This document provides a comprehensive comparison of V3 feature implementation status across Apache Iceberg Core, Apache Spark, and Apache Flink.
+
+## V3 Features Implementation Status
+
+| **V3 Feature** | **Core** | **Spark v3.4** | **Spark v3.5** | **Spark v4.0** | **Flink v1.20** | **Flink v2.0** | **Flink v2.1** |
+|---|---|---|---|---|---|---|---|
+| **New Data Types** | | | | | | | |
+| `unknown` type | ✅ | ❌ | ❌ | ✅ | ✅ | ✅ | ✅ |
+| `timestamp_ns` (nanosecond timestamps) | ✅ | ❌ | ❌ | ❌ | ✅ | ✅ | ✅ |
+| `variant` type | ⚠️ Avro supported, Parquet supported, ORC not supported | ⚠️ Read supported, Write shredded not supported | ⚠️ Read supported, Write shredded not supported | ⚠️ Read supported, Write shredded not supported | ❌ | ❌ | ❌ |
+| `geometry` type | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
+| `geography` type | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
+| **Deletion Vectors** | | | | | | | |
+| Binary Deletion vector support | ✅ | ❌ | ❌ | ✅ | ❌ | ❌ | ❌ |
+| **Default Values** | | | | | | | |
+| Field default values | ✅ | ⚠️ Read/Write supported, DDL not supported | ⚠️ Read/Write supported, DDL not supported | ⚠️ Read/Write supported, DDL not supported | ✅ | ✅ | ✅ |
+| **Table Encryption** | | | | | | | |
+| AES-GCM encryption | ✅ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
+| **Multi-Argument Transforms** | | | | | | | |
+| Z-order partitioning/sorting | ✅ | ✅ | ✅ | ✅ | ❌ | ❌ | ❌ |
+| **Row Lineage** | | | | | | | |
+| `_row_id` column | ✅ | ✅ | ✅ | ✅ | ❌ | ❌ | ❌ |
+| `_last_updated_sequence_number` | ✅ | ✅ | ✅ | ✅ | ❌ | ❌ | ❌ |
```

Comment:
I feel like row lineage should just be a single row. Separating row_id/seq number columns doesn't make too much sense since handling both is required to even support row lineage in general.

Reply:
Done